elm/web/rhub.py

```diff
@@ -4,626 +4,788 @@
+            full += research
 
-        return logger.info('Finished processing publications')
+        if experience:
+            research = (f"{name}'s professional experience includes: "
+                        f"{experience}. ")
+            full += research
 
+        if self.education:
+            for edu in self.education:
+                full += edu
 
-class ResearcherProfiles():
-    """
-    Class to handle researcher profiles portion of the NREL researcher hub.
-    """
-    BASE_URL = "https://research-hub.nrel.gov/en/persons/?page=0"
+        if self.publications:
+            publications = (f"{name} has been involved in the following "
+                            f"publications: {', '.join(self.publications)}. ")
+            full += publications
 
-    def __init__(self, url, n_pages=1, txt_dir='./ew_txt'):
+        with open(fp, "w") as text_file:
+            text_file.write(full)
+
+
+class ProfilesList(list):
+    """Class to retrieve and handle multiple profiles from an API URL."""
+    def __init__(self, url, n_pages=1):
         """
         Parameters
         ----------
         url : str
-            Research hub profiles URL, most likely
-            https://research-hub.nrel.gov/en/persons/
+            Research Hub API URL to request, see this for details:
+                https://research-hub.nrel.gov/ws/api/524/api-docs/index.html
         n_pages : int
-            Number of pages to get from the API. Typical response has 50
+            Number of pages to get from the API. Typical response has 20
             entries per page. Default of 1 ensures that this class doesnt hang
             on a million responses.
-        txt_dir : str
-            File directory where you would like to save output .txt files.
         """
+        api_key = os.getenv("RHUB_API_KEY")
+        assert api_key is not None, "Please set RHUB_API_KEY!"
 
-        self.text_dir = txt_dir
-        self.profile_links = []
-        for p in range(0, n_pages):
-            url_base = url + f"?page={p}"
-            with urlopen(url_base) as page:
-                html = page.read().decode("utf-8")
-            soup = BeautifulSoup(html, "html.parser")
-
-            target = soup.find('ul', {'class': 'grid-results'})
-            docs = target.find_all('a', {'class': 'link'})
+        self.url = url
+        self._session = requests.Session()
+        self._response = None
+        self._n_pages = 0
+        self._iter = 0
 
-            page_links = [d['href'] for d in docs if '/persons/' in d['href']]
-            self.profile_links.extend(page_links)
+        records = self._get_first()
+        for page in self._get_pages(n_pages=n_pages):
+            records += page
+        records = [ProfilesRecord(single) for single in records]
+        super().__init__(records)
 
-    def build_meta(self):
-        """Build a meta dataframe containing relevant information for
-        researchers.
+    def _get_first(self):
```

Comment:
Recommend wrapping _get_first and _get_pages in a single function so it's easy to _get_all()

You could do this in the osti module too (i realize you're just copying my mediocre implementation).